src/plugins/intel_cpu/src/nodes/multinomial.cpp

```diff
@@ -0,0 +1,200 @@
+    m_batches_count = probs_shape.size() == 2 ? probs_shape[0] : 1;
+    m_probs_count = probs_shape.size() == 2 ? probs_shape[1] : probs_shape[0];
+
+    m_cdf.reserve(m_input_elements_count);
+    m_input_vals.reserve(m_input_elements_count);
+    m_random_samples.reserve(m_input_elements_count);
+    m_max_per_batch.reserve(probs_shape.size() == 1 ? 1 : probs_shape[1]);
+}
+
+void Multinomial::execute(dnnl::stream strm) {
+    const auto* probs = reinterpret_cast<const float*>(getParentEdgeAt(PROBS_PORT)->getMemoryPtr()->getData());
+    const auto num_samples =
+        reinterpret_cast<const int*>(getParentEdgeAt(NUM_SAMPLES_PORT)->getMemoryPtr()->getData())[0];
+    auto* output = reinterpret_cast<int*>(getChildEdgeAt(OUTPUT_PORT)->getMemoryPtr()->getData());
+
+    // exp & cumsum
+    if (m_log_probs) {
+        parallel_for(m_input_elements_count, [&](size_t idx) {
+            m_input_vals[idx] = std::exp(probs[idx]);
+        });
+        m_cdf[0] = m_input_vals[0];
+        for (size_t idx = 1; idx < m_input_elements_count; ++idx) {
+            m_cdf[idx] = m_input_vals[idx] + m_cdf[idx - 1];
+        }
+    } else {
+        m_cdf[0] = probs[0];
+        for (size_t idx = 1; idx < m_input_elements_count; ++idx) {
+            m_cdf[idx] = probs[idx] + m_cdf[idx - 1];
+        }
+    }
+
+    // TODO RandomUniform - requires RandomUniform kernel https://github.com/openvinotoolkit/openvino/pull/20171
+    std::srand(m_op_seed);
+    parallel_for(m_input_elements_count, [&](size_t idx) {
+        m_random_samples[idx] = static_cast<double>(std::rand()) / static_cast<double>(RAND_MAX);
+    });
+
+    // max (slice) & divide
+    if (m_probs_1d) {
+        m_max_per_batch[0] = m_cdf[m_input_elements_count - 1];
+        parallel_for(m_input_elements_count, [&](size_t idx) {
+            m_cdf[idx] /= m_max_per_batch[0];
+        });
+    } else {
+        parallel_for(m_batches_count, [&](size_t idx) {
+            m_max_per_batch[idx] = m_cdf[idx * m_batches_count - 1];
+        });
+        parallel_for(m_input_elements_count, [&](size_t idx) {
+            size_t idx_max_elem = idx / m_probs_count;
+            m_cdf[idx] /= m_max_per_batch[idx_max_elem];
+        });
+    }
+
+    if (m_with_replacement) {
+        parallel_for(m_input_elements_count, [&](size_t idx) {
+            size_t idx_output = idx / m_probs_count * num_samples;
+            size_t idx_class = idx % m_probs_count;
+            if (m_random_samples[idx] <= m_cdf[idx] && (!idx_class || m_random_samples[idx] > m_cdf[idx - 1])) {
+                output[idx_output] = static_cast<int>(idx_class);
+            }
+        });
+    } else {
+        parallel_for(m_input_elements_count, [&](size_t idx) {
+            size_t idx_batch = idx / m_probs_count;
+            size_t idx_class = idx % m_probs_count;
+            size_t idx_output = idx_batch * num_samples;
+
+            float class_probability = 0.0f;
+            float divisor = 0.0f;
```

Comment:
That is not applicable value for divisor. Set non-zero value to avoid runtime exception.

Reply:
Great suggestion, set to 1.0f!